src/core/unit.c

```diff
@@ -845,6 +867,9 @@ Unit* unit_free(Unit *u) {
         if (u->in_release_resources_queue)
                 LIST_REMOVE(release_resources_queue, u->manager->release_resources_queue, u);
 
+        if (u->in_stop_notify_queue)
+                LIST_REMOVE(stop_notify_queue, u->manager->stop_notify_queue, u);
```

Comment:
Why not to call unit_remove_from_stop_notify_queue() ?

Reply:
To make things consistent with other queues above? Plus no need to reset the flag here

Reply:
If there is a specific cleanup function, then we should call it, even if it does something redundant on freeing.